src/System.Runtime.Handles/tests/SafeHandle.cs

```diff
@@ -81,4 +81,40 @@ public static void SafeHandle_valid_close()
         Assert.False(mch.IsInvalid);
         Assert.True(mch.IsReleased);
     }
+
+    [DllImport("Kernel32", SetLastError = true)]
+    private extern static void SetLastError(int error);
+
+    private class LastErrorSafeHandle : SafeHandle
+    {
+        internal LastErrorSafeHandle(IntPtr h)
+            : base(h, true)
+        {
+        }
+
+        public override bool IsInvalid => handle == IntPtr.Zero;
+
+        override protected bool ReleaseHandle()
+        {
+            SetLastError(-1);
+            return true;
+        }
+    }
+
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
On Unix, it would have to use private reflection or depend on System Native shim.